### PR TITLE
Custom offline page service worker revamp

### PR DESCRIPTION
**Pull request by jeffposnick**
I still refer folks to this periodically, and I realize that it could really use a revamp; modern JavaScript, and, in particular, illustrate the use of navigation preload when the browser supports it.

### CODE REVIEW

**Review by beaufortfrancois — APPROVED** [review]
LGTM

File: service-worker/custom-offline-page/service-worker.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- Copyright 2015 Google Inc. All Rights Reserved.
+ Copyright 2019 Google Inc. All Rights Reserved.
```
Comment (tomayac) [inline service-worker/custom-offline-page/service-worker.js]: If you update the copyright year, you give up the copyright before the updated year. If the code has substantially changed, the recommendation is to add a new year:

 ```suggestion
 Copyright 2015, 2019 Google Inc. All Rights Reserved.
```

File: service-worker/custom-offline-page/service-worker.js [hunk 2]
```diff
@@ -11,92 +11,66 @@
...
-    fetch(createCacheBustedRequest(OFFLINE_URL)).then(function(response) {
-      return caches.open(CURRENT_CACHES.offline).then(function(cache) {
-        return cache.put(OFFLINE_URL, response);
-      });
-    })
-  );
+self.addEventListener('install', (event) => {
+  event.waitUntil((async () => {
+    const cache = await caches.open(CACHE_NAME);
+    // Setting cache: reload in the new request will ensure that the response
```
Comment (tomayac) [inline service-worker/custom-offline-page/service-worker.js]: The 'reload' value is a string: https://developer.mozilla.org/en-US/docs/Web/API/Request/cache#Value.

```suggestion
    // Setting {cache: 'reload'} in the new request will ensure that the response
```

File: service-worker/custom-offline-page/service-worker.js [hunk 3]
```diff
@@ -11,92 +11,66 @@
...
+        if (preloadResponse) {
+          return preloadResponse;
+        }
+
+        const networkResponse = await fetch(event.request);
+        return networkResponse;
+      } catch (error) {
+        // catch is only triggered if an exception is thrown, which is likely
+        // due to a network error.
+        // If fetch() returns a valid HTTP response with an response code in
```
Comment (tomayac) [inline service-worker/custom-offline-page/service-worker.js]: Typo:

```suggestion
        // If fetch() returns a valid HTTP response with a response code in
```

**Review by tomayac — APPROVED** [review]
LGTM2, with three minor nits.

File: service-worker/custom-offline-page/service-worker.js (hunk 1 above)
Comment (jeffposnick) [reply service-worker/custom-offline-page/service-worker.js]: Gotcha.

**Pull request merged**